pkg/detectors/uri/uri.go

```diff
@@ -23,7 +23,7 @@ var _ detectors.Detector = (*Scanner)(nil)
 var _ detectors.CustomFalsePositiveChecker = (*Scanner)(nil)
 
 var (
-	keyPat = regexp.MustCompile(`\b(?:https?:)?\/\/[\S]{3,50}:([\S]{3,50})@[-.%\w\/:]+\b`)
+	keyPat = regexp.MustCompile(`\b(?:https?:)?\/\/[\w-\.]{3,50}:([\w-\.]{3,50})@[-.%\w\/:]+\b`)
```

Comment:
In my opinion, passwords in URIs can consist of any non-whitespace characters. Using \w limits the match to a specific set of characters, which might exclude valid ones. Are you noticing any detection issues when using \S instead?

Reply:
e.g. [http://username:p%40ssword@127.0.0.1"](http://username:p%40ssword@127.0.0.1) Will not be detected when using `\w`.

Reply:
Yes, actually as the scenario mentioned in the linked issue, it matches some special chars like `"` which it should not I guess. I can add more special characters like `%$^...` in the set to expand the functionality.

Reply:
I add some more possible special characters for password